### ISSUE TITLE
[Event Request] - Codeunit CRM Sales Order to Sales Order - OnBeforeSetLineDescription

### ISSUE THREAD

**Issue by DanielHaglo** [microsoft/ALAppExtensions#12617]
![image](https://user-images.githubusercontent.com/35260264/120295446-6c870300-c2c7-11eb-8a6e-0d1420c6a1e9.png)

Can you create this event?

**Comment by Oleg-Romashkov** [comment]
Please always put the full detailed signature of the requested events and the parameters passed in the call.
You help us implementing the requests faster when your request contains AL code copied from VSCode instead of screenshots.

**Comment by DanielHaglo** [comment]
Hi Oleg,

My bad. Let me try again.

 local procedure SetLineDescription(SalesHeader: Record "Sales Header"; var SalesLine: Record "Sales Line"; var CRMSalesorderdetail: Record "CRM Salesorderdetail")
    var
        SalesReceivablesSetup: Record "Sales & Receivables Setup";
        LineDescriptionInStream: InStream;
        CRMSalesOrderLineDescription: Text;
        ExtendedDescription: Text;
        IsHandled: Boolean;
    begin
        //Event
        OnBeforeSetLineDescription(SalesLine, CRMSalesorderdetail, IsHandled);
        if IsHandled then
            exit;
        //EVent
        CRMSalesorderdetail.CalcFields(Description);
        CRMSalesorderdetail.Description.CreateInStream(LineDescriptionInStream, TEXTENCODING::UTF16);
        LineDescriptionInStream.ReadText(CRMSalesOrderLineDescription);

        if CRMSalesOrderLineDescription = '' then
            CRMSalesOrderLineDescription := CRMSalesorderdetail.ProductDescription;

        ExtendedDescription := CRMSalesOrderLineDescription;

        // in case of write-in product - write the description directly in the main sales line description
        if SalesReceivablesSetup.Get() then
            if SalesLine."No." = SalesReceivablesSetup."Write-in Product No." then begin
                SalesLine.Description := CopyStr(CRMSalesOrderLineDescription, 1, MaxStrLen(SalesLine.Description));
                if StrLen(CRMSalesOrderLineDescription) > MaxStrLen(SalesLine.Description) then
                    ExtendedDescription := CopyStr(CRMSalesOrderLineDescription, MaxStrLen(SalesLine.Description) + 1)
                else
                    ExtendedDescription := '';
            end;

        if ExtendedDescription = SalesLine.Description then
            exit;

        // in case of inventory item - write the item name in the main line and create extended lines with the extended description
        CreateExtendedDescriptionOrderLines(SalesHeader, ExtendedDescription, SalesLine."Line No.");

        // in case of line descriptions with multple lines, add all lines of the line descirption
        while not LineDescriptionInStream.EOS() do begin
            LineDescriptionInStream.ReadText(ExtendedDescription);
            CreateExtendedDescriptionOrderLines(SalesHeader, ExtendedDescription, SalesLine."Line No.");
        end;
    end;

// ----- Event --------//

[IntegrationEvent(false, false)]
    local procedure OnBeforeSetLineDescription(var SalesLine: Record "Sales Line"; var CRMSalesOrderDetail: Record "CRM Salesorderdetail"; var IsHandled: Boolean)
    begin
    end;

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.
